vsintegration/src/FSharp.Editor/CodeLens/CodeLens.fs

```diff
@@ -0,0 +1,799 @@
+    //            if newLine <> lineNumber then
+    //                trackingSpans.[lineNumber].Remove(trackingSpan) |> ignore
+    //                trackingSpans.[newLine].Add(trackingSpan)
+    
+    let updateTrackingSpansFast (snapshot:ITextSnapshot) lineNumber =
+        if lineNumber |> trackingSpans.ContainsKey then
+            let currentTrackingSpans = Generic.List(trackingSpans.[lineNumber])
+            for trackingSpan in currentTrackingSpans do
+                let newLineOption = tryGetTSpanStartLine snapshot trackingSpan
+                match newLineOption with 
+                | None -> ()
+                | Some newLine ->
+                    if newLine <> lineNumber then
+                        trackingSpans.[lineNumber].Remove(trackingSpan) |> ignore
+                        if newLine |> trackingSpans.ContainsKey |> not then
+                            trackingSpans.[newLine] <- Generic.List()
+                        trackingSpans.[newLine].Add(trackingSpan)
+                        //tagsChangedEvent.Trigger(self, SnapshotSpanEventArgs(trackingSpan.GetSpan snapshot))
+
+    let createDefaultStackPanel () = 
+        let uiElement = new StackPanel()
+        uiElement.Visibility <- Visibility.Hidden
+        uiElement
+
+    /// Helper methods which invokes every action which is needed for new trackingSpans
+    let addTrackingSpan trackingSpan =
+        let snapshot = buffer.CurrentSnapshot
+        let startLineNumber = getTrackingSpanStartLine snapshot trackingSpan
+        if trackingSpans.ContainsKey(startLineNumber) then
+            trackingSpans.[startLineNumber].Add trackingSpan
+        else
+            trackingSpans.[startLineNumber] <- Generic.List()
+            trackingSpans.[startLineNumber].Add trackingSpan
+        uiElements.[trackingSpan] <- createDefaultStackPanel ()
+        tagsChangedEvent.Trigger(self, SnapshotSpanEventArgs(trackingSpan.GetSpan(snapshot)))
+        uiElements.[trackingSpan]
+    
+    /// Layouts all stack panels on the line
+    let layoutUIElementOnLine (view:IWpfTextView) (line:SnapshotSpan) (ui:UIElement) =
+        try
+            // Get the real offset so that the code lens are placed respectively to their content
+            let offset =
+                [0..line.Length - 1] |> Seq.tryFind (fun i -> not (Char.IsWhiteSpace (line.Start.Add(i).GetChar())))
+                |> Option.defaultValue 0
+            let realStart = line.Start.Add(offset)
+            // Get the geometry which respects the changed height due to the SpaceAdornmentTag
+            let geometry = 
+                let g = view.TextViewLines.GetCharacterBounds(realStart)
+                // WORKAROUND VS BUG, left cannot be zero if the offset is creater than zero!
+                // Calling the method twice fixes this bug and ensures that all values are correct.
+                if g.Left = 0. && offset > 0 then view.TextViewLines.GetCharacterBounds(realStart)
```

Comment:
Has this bug been reported?

Reply:
No, I thought the people from the Visual Studio Team wouldn't care anyways. 
Also I don't know whether it's worth to report to them, however I can tell that it's an annoying bug for me. There's no other way to receive the offset :-(